### PR TITLE
Implemented JsonRpcMarshaller

### LINKED ISSUE #13: Create JSON (De)Marshaller for RPCs
The JsonRpcMarshaller is responsible for converting messages to and from byte arrays and needs to be implemented before being able to sending and receiving RPC messages.

### PR DESCRIPTION
**Pull request by russjohnson09**
Fixes #13 

This PR is **ready** for review.

### Risk
This PR makes **major** API changes.

### Testing Plan
Send an rpcMessage through the marshall and convert it back to a json object through the unmarshall function. Confirm that the output matches the parameters given to the rpc message.

### Summary
Implements the JsonRpcMarshaller.

### CLA
- [x ] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform) and this pull request adheres to the [Contributing Guide](./CONTRIBUTING.md).

### CODE REVIEW

**Comment by crokita** [comment]
Should RpcMessage be RpcStruct for what the marshall method can accept?

**Comment by russjohnson09** [comment]
> Should RpcMessage be RpcStruct for what the marshall method can accept?
marshall requires an object that defines getParameters which is defined on the RpcStruct class extended by RpcMessage. I updated the jsdoc to be type RpcStruct.

File: lib/js/util/JsonRpcMarshaller.js [hunk 1]
```diff
@@ -33,62 +33,71 @@
...
+        return ret;
+    }
+
+    /**
+     * Transform a string into a Uint8Array.
+     * @param {String} str - String to transform.
+     * @returns {Uint8Array}
+     * @private
+     */
+    static _encode(str) {
```
Comment (russjohnson09) [inline lib/js/util/JsonRpcMarshaller.js]: This will not work for utf-8 characters exceeding 255. eg "移".charCodeAt() = 31227

Comment (russjohnson09) [reply lib/js/util/JsonRpcMarshaller.js]: https://developer.mozilla.org/en-US/docs/Web/API/TextEncoder#Polyfill Using this function that will work for both node and browser based javascript.

File: lib/js/src/util/TextEncoder.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
+class TextEncoder {
+    // https://developer.mozilla.org/en-US/docs/Web/API/TextEncoder#Polyfill
+    encode (str) {
```
Comment (nickschwab) [inline lib/js/src/util/TextEncoder.js]: Looks like it doesn't make sense to require an instance of the class to access the `encode()` method. Recommend changing to:
```suggestion
    static encode (str) {
```

File: lib/js/src/util/JsonRpcMarshaller.js [hunk 3]
```diff
@@ -30,64 +30,68 @@
...
     }
 
+    /**
+     * Transform a string into a byte array.
+     * @param {String} str - String to transform.
+     * @returns {Array}
+     * @private
+     */
+    static _encode (str) {
+        const encoder = new TextEncoder();
```
Comment (nickschwab) [inline lib/js/src/util/JsonRpcMarshaller.js]: Creating a new instance of `TextEncoder` probably doesn't make sense. See other feedback regarding making the TextEncoder's `encode()` method static.

File: lib/js/src/util/JsonRpcMarshaller.js [hunk 4]
```diff
@@ -30,64 +30,68 @@
...
-            let jsonObject = rpcMessage.getParameters();
-            //TODO do we need to go through each param and do this?
-            let stringVersion = JSON.stringify(jsonObject);
-            let encoder = new TextEncoder();
-            return encoder.encode(stringVersion);
-
-        } catch (exception) {
-            //TODO print exception
+    /**
+     * Takes an RPC message and converts it
```
Comment (nickschwab) [inline lib/js/src/util/JsonRpcMarshaller.js]: Incorrect documentation?

File: lib/js/src/util/TextEncoder.js (hunk 2 above)
Comment (russjohnson09) [reply lib/js/src/util/TextEncoder.js]: updated to static

File: lib/js/src/util/JsonRpcMarshaller.js (hunk 3 above)
Comment (russjohnson09) [reply lib/js/src/util/JsonRpcMarshaller.js]: updated

File: lib/js/src/util/JsonRpcMarshaller.js (hunk 4 above)
Comment (russjohnson09) [reply lib/js/src/util/JsonRpcMarshaller.js]: fixed duplicate documentation

**Pull request merged**